gpflow/kernels/conditioned.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+import tensorflow as tf
+from .base import Kernel
+from ..config import default_float, default_jitter
+
+
+class Conditioned(Kernel):
+    """
+    Conditioned kernel. Can be used to wrap any kernel
+    to transform it into a its conditioned version.
+
+    This provides a simple way to condition a Gaussian process on noiseless observations
+
+    From a covariance function k(.,.) and a conditioning dataset (Xc, Yc),
+    a new covariance function and a conditional mean function
+    are created as
+
+       kc(.,.) = k(.,.) - k(.,Xc)k(Xc, Xc)⁻¹k(Xc,.)
+       mc(.)   = k(.,Xc)k(Xc, Xc)⁻¹Yc
+    """
+
+    def __init__(self, base: Kernel, xc: tf.Tensor, yc: tf.Tensor):
+        """
+        :param base: the base kernel to make conditioned; must inherit from Kernel
+        :param xc: conditioning input
+        :param yc: conditioning output
+        """
+
+        if not isinstance(base, Kernel):
+            raise TypeError("Conditioned requires a Kernel object as the `base`")
+
+        super().__init__()
+        self.base = base
+        self.xc = xc
+        self.yc = yc
+        self.num_c = xc.shape[0]
+
+    @property
+    def Lc(self):
```

Comment:
The naming for this function can be improved. Can you come up with another name?

Reply:
renamed!